src/app/libp2p_helper/src/libp2p_helper/main.go

```diff
@@ -33,6 +33,38 @@ import (
 	"github.com/multiformats/go-multiaddr"
 )
 
+// a grow-only dynamically sized peer set (because generics are for nerds)
```

Comment:
there's a `peer.Set` exposed that we can use: https://godoc.org/github.com/libp2p/go-libp2p-core/peer#Set

Reply:
I didn't realize that go slices are backed by both a logical size and a capacity when I wrote this, and that you can call a 3 arity version of `make` to set both separately. Given that, I'm just going to opt to fix the original code to allocate `AddedPeers` as `make([]peer.Addr, 0, XXX)` and use `append` instead.